pkg/elevate/elevate.go

```diff
@@ -0,0 +1,225 @@
+
+	err := cmd.Run()
+	if err != nil {
+		return "", err
+	}
+	username := strings.TrimSpace(buf.String())
+	if username == "" {
+		return "", errors.New("Username cannot be empty")
+	}
+
+	return username, nil
+}
+
+func AddImpersonationReasonRoleToCurrentUser(e ElevateConfig, config api.Config) error {
+	clientCfg, err := BuildConfigFromFlags("", clientcmd.NewDefaultPathOptions().GetDefaultFilename())
+	if err != nil {
+		return err
+	}
+	clientCfg.Impersonate.UserName = ImpersonationUser
+
+	clientSet, err := NewClientset(clientCfg)
+	if err != nil {
+		return err
+	}
+	ctx := context.TODO()
+
+	err = clientSet.RbacV1().ClusterRoles().Delete(ctx, ElevateBackplaneReasonClusterRole, metav1.DeleteOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+
+	err = clientSet.RbacV1().ClusterRoleBindings().Delete(ctx, ElevateBackplaneReasonClusterRole, metav1.DeleteOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+
+	role, err := clientSet.RbacV1().ClusterRoles().Create(
+		ctx,
+		&v1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: ElevateBackplaneReasonClusterRole,
+			},
+			Rules: []v1.PolicyRule{
+				{
+					Verbs:     []string{"impersonate"},
+					APIGroups: []string{"authentication.k8s.io"},
+					Resources: []string{"userextras/reason"},
+				},
+			},
+		},
+		metav1.CreateOptions{},
+	)
+	if err != nil {
+		return err
+	}
+
+	username, err := e.GetCurrentUsername(e)
+	if err != nil {
+		return err
+	}
+
+	_, err = clientSet.RbacV1().ClusterRoleBindings().Create(
+		ctx,
+		&v1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: ElevateBackplaneReasonClusterRole,
+			},
+			RoleRef: v1.RoleRef{
+				APIGroup: clientSet.RbacV1().RESTClient().APIVersion().Group,
+				Kind:     "ClusterRole",
+				Name:     role.Name,
+			},
+			Subjects: []v1.Subject{
+				{
+					Kind:     "User",
+					APIGroup: clientSet.RbacV1().RESTClient().APIVersion().Group,
+					Name:     username,
+				},
+			},
+		},
+		metav1.CreateOptions{},
+	)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func AddElevationReasonToRawKubeconfig(config api.Config, elevationReason string) error {
+	logger.Debugln("Adding reason for backplane-cluster-admin elevation")
+	if config.Contexts[config.CurrentContext] == nil {
+		return errors.New("No current kubeconfig context")
+	}
+
+	currentCtxUsername := config.Contexts[config.CurrentContext].AuthInfo
+
+	if config.AuthInfos[currentCtxUsername] == nil {
+		return errors.New("No current user information")
+	}
+
+	if config.AuthInfos[currentCtxUsername].ImpersonateUserExtra == nil {
+		config.AuthInfos[currentCtxUsername].ImpersonateUserExtra = make(map[string][]string)
+	}
+
+	config.AuthInfos[currentCtxUsername].ImpersonateUserExtra["reason"] = []string{elevationReason}
+	config.AuthInfos[currentCtxUsername].Impersonate = "backplane-cluster-admin"
+
+	return nil
+}
+
+func WriteConfigToTempKubeconfig(config api.Config) (string, error) {
```

Comment:
You may be able to use this func
https://github.com/openshift/backplane-cli/blob/34e4c2c59fa2af60371ce955ab4e089755bb5a68/pkg/utils/util.go#L162